### PR TITLE
Allow customizing ciphers and other OpenSSL settings

### PR DESCRIPTION
**Pull request by Joannis**
To do:

- Docs
- Support for more settings

### CODE REVIEW

File: Sources/OpenSSL/OpenSSLCiphers.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+public indirect enum OpenSSLCipher {
```
Comment (tanner0101) [inline Sources/OpenSSL/OpenSSLCiphers.swift:1]: Comments

File: Sources/OpenSSL/OpenSSLClient.swift [hunk 2]
```diff
@@ -15,8 +15,12 @@ public final class OpenSSLClient: TLSClient {
     private let tcp: TCPClient
 
     /// Create a new `OpenSSLClient`
-    public init(tcp: TCPClient, using settings: TLSClientSettings) throws {
-        let socket = try OpenSSLSocket(tcp: tcp.socket, method: .tls1_2, side: .client)
+    public init(
+        tcp: TCPClient,
+        using settings: TLSClientSettings,
+        configuration: OpenSSLSettings = OpenSSLSettings()
```
Comment (tanner0101) [inline Sources/OpenSSL/OpenSSLClient.swift:21]: the double settings thing here is quite confusing. Maybe we can come up w/ a different name? Or perhaps `TLSClientSettings` can have a generic parameter `custom: T` which stores additional custom settings?

**Review by tanner0101 — CHANGES_REQUESTED** [review]
the double settings thing is a bit confusing, but otherwise looks good!

File: Sources/OpenSSL/OpenSSLClient.swift (hunk 2 above)
Comment (Joannis) [reply Sources/OpenSSL/OpenSSLClient.swift:21]: That's my problem, too. I don't really know how.

I'll add comments, but I'm not sure if I can easily add tests here.

**Pull request closed without merging**